packages/react/src/components/diffViewer/DiffViewer.tsx

```diff
@@ -1,22 +1,55 @@
-import {ComponentProps, FC} from 'react';
-import ReactDiffViewer, {ReactDiffViewerStylesOverride} from 'react-diff-viewer';
+import {FunctionComponent, ReactNode} from 'react';
+import {parseDiff, Decoration, Diff, Hunk} from 'react-diff-view';
 
 import {BlankSlate} from '../blankSlate';
 
-/**
- * @deprecated Use oldValue and newValue Props.
- */
-export interface DiffViewerPropsDeprecated {
-    /**
-     * @deprecated Use oldValue instead.
-     */
-    first: string;
-    /**
-     * @deprecated Use newValue instead.
-     */
-    second: string;
+enum DIFF_VIEWER_VIEW_TYPE {
+    SPLIT = 'split',
+    UNIFIED = 'unified',
+}
+
+interface Hunk {
+    oldStart: number;
+    oldLines: number;
+    newStart: number;
+    newLines: number;
+    content: string;
+    changes: Array<{
+        type: 'delete' | 'insert' | 'normal';
+        content: string | ReactNode;
+        isNormal?: boolean;
+        isInsert?: boolean;
+        isDelete?: boolean;
+        lineNumber?: number;
+        oldLineNumber?: number;
+        newLineNumber?: number;
+    }>;
+}
+
+interface FileProps {
+    hunks: Hunk[];
+    newRevision: string;
+    oldRevision: string;
+    type: 'add' | 'delete' | 'modify' | 'rename' | 'copy';
+    viewType: DIFF_VIEWER_VIEW_TYPE;
 }
 
+const File: FunctionComponent<FileProps> = ({oldRevision, newRevision, hunks, type, viewType}) => (
```

Comment:
The name `File` already exists in the global namespace, maybe we should give it a different name.